src/AutoRest.CSharp.V3/Pipeline/Extensions.cs

```diff
@@ -85,102 +83,142 @@ internal static class Extensions
             { typeof(Uri), n => null } //TODO: Figure out how to get the Uri type here, so we can do 'new Uri(GetString())'
         };
 
+        private static void WriteSerializeClientObject(WriterBase writer, string name, bool isNullable)
+        {
+            writer.Append(name);
+            writer.Append(isNullable ? "?" : string.Empty);
+            writer.Line(".Serialize(writer);");
+        }
+
+        private static void WriteSerializeClientEnum(WriterBase writer, string name, bool isNullable, bool isStringBased)
+        {
+            writer.Append("writer.WriteStringValue(");
+            writer.Append(name);
+            writer.Append((!isStringBased && isNullable) ? ".Value" : string.Empty);
```

Comment:
I don't like this pattern. If the alternative is not to do anything why not use if?

Reply:
It is data driven design over logic driven design. I'll change this to be an if-statement.